module/executiondatasync/execution_data/processed_height_recorder.go

```diff
@@ -1,24 +1,34 @@
 package execution_data
 
 import (
+	"sync"
+
 	"go.uber.org/atomic"
 )
 
+type HeightUpdatesConsumer func(height uint64)
+
 // ProcessedHeightRecorder is an interface for tracking the highest execution data processed
 // height when a block is processed and for providing this height.
 type ProcessedHeightRecorder interface {
 	// OnBlockProcessed updates the highest processed height when a block is processed.
 	OnBlockProcessed(uint64)
 	// HighestCompleteHeight returns the highest complete processed block height.
 	HighestCompleteHeight() uint64
+
+	// AddHeightUpdatesConsumer subscribe consumer for processed height updates.
+	// Callback are called synchronously and must be non-blocking
+	AddHeightUpdatesConsumer(HeightUpdatesConsumer)
 }
 
 var _ ProcessedHeightRecorder = (*ProcessedHeightRecorderManager)(nil)
 
 // ProcessedHeightRecorderManager manages an execution data height recorder
 // and tracks the highest processed block height.
 type ProcessedHeightRecorderManager struct {
+	sync.Mutex
```

Comment:
I don't think we want to expose the mutex interface to external callers. best to use an unexported variable